### PR TITLE
Add tools/atp-extract.py

### PR DESCRIPTION
**Pull request by hyperlogic**
Tool to list and extract files from the atp assets directory.

Usage: python2 atp-extract.py -[lxa] [filename]

cd into the `c:\Users\BettySpaghetti\AppData\Roaming\High Fidelity\assignment-client\assets` directory

* run `python2 atp-extract.py -l` to list all files
* run `python2 atp-extract.py -x file` to extract that particular file to the current directory.
* run `python2 atp-extract.py -a` to extract all files.

### CODE REVIEW

**Comment by hyperlogic** [comment]
QA test:
* install python2
* open a command prompt
* cd into `c:\Users\BettySpaghetti\AppData\Roaming\High Fidelity\assignment-client\assets`
* test `python2 atp-extract.py -l` it should list all files in the atp cache.
* test `python2 atp-extract.py -x file` it should extract a particular file to the current directory.
* test `python2 atp-extract.py -a` extracts all files into the current directory

**Comment by hifi-gustavo** [comment]
Build console output for commit: 38a2ce67bbeb7af6be716f1d46716fdbcf96e5bc available at: https://gist.github.com/805b51a96d7d769ca1798f27c597255f<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8999-38a2ce6.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8999-38a2ce6.dmg

File: tools/atp-extract.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+elif option == '-x':
+    assetMap = loadMapFile("map.json")
+    outputFilename = sys.argv[2]
+    if not extractFile(assetMap, outputFilename):
+        print("Error could not extract file: \"" + outputFilename + "\"")
+elif option == '-a':
+    assetMap = loadMapFile("map.json")
+    for key, value in assetMap.iteritems():
+        print("Extracting " + key[1:])
+        extractFile(assetMap, key[1:])
```
Comment (huffman) [inline tools/atp-extract.py]: Should an error be printed here on error?

**Comment by huffman** [comment]
This is great 👍

File: tools/atp-extract.py (hunk 1 above)
Comment (hyperlogic) [inline tools/atp-extract.py]: This would only fail of map.json has an empty entry.

**Comment by OmegaHeron** [comment]
I suspect this is an advanced power user/internal use tool, but, that said, it works fine from my linux box.

**Pull request merged**